### ISSUE TITLE
Fine-Tuning TokenClassification NER Bert Model

### ISSUE THREAD

**Issue by rajatjaiswal96** [NVIDIA/NeMo#4296]
**Describe the bug**

**Steps/Code to reproduce bug**

While doing the Fine-Tuning of pretrained ner_en_bert model, Model Config file is not getting update with the new one. It is using the same config file on which model is pretrained on. I can see the same in logs after calling trainer.fit()
I've already tried with below settings:

model.cfg = cfg
model._cfg = cfg
model._hparams_initial = cfg

    model_from_scratch = nemo_nlp.models.TokenClassificationModel.from_pretrained("ner_en_bert")
    cfg = copy.deepcopy(model_from_scratch.cfg)
    cfg.dataset.data_dir = data_dir
    cfg.label_ids = {'O': 0, 'B-place': 1, 'B-emotion': 2, 'B-noun': 3, 'B-name': 4}
    print('BEFORE:',cfg)
    cfg.optim.lr = 0.1
    print('CFG',cfg)
    model_from_scratch.cfg.label_ids = {'O': 0,'B-place':1,'B-emotion':2,'B-noun':3,'B-name':4}
    model_from_scratch.cfg.dataset.data_dir = data_dir #passed using input function parameters
    model_from_scratch.update_data_dir(data_dir)
    model_from_scratch.setup_training_data()
    model_from_scratch.setup_validation_data()
    trainer = pl.Trainer(devices=1, accelerator='cpu',max_epochs=2, log_every_n_steps =1,max_steps = 32)
    model_from_scratch.set_trainer(trainer)
    trainer.fit(model_from_scratch)

**Expected behavior**

Model Should get trained with the new config changes

**Comment by ericharper** [comment]
@ekmb could you look at this or assign it to someone who can?

**Comment by ekmb** [comment]
@rajatjaiswal96 this is the expected behavior: either finetune the model on a dataset that has the same set of labels the original model was trained on (see the old config) or start BERT fine-tuning from scratch (don't load pre-trained ner model).
